elements/nuxeo-app.html

```diff
@@ -1160,36 +1160,50 @@
       },
 
       _computeEnrichers: function() {
-        return (Nuxeo.UI && Nuxeo.UI.config && Nuxeo.UI.config.enrichers && Nuxeo.UI.config.enrichers.join(',')) ||
-          [
-            'hasContent',
-            'firstAccessibleAncestor',
-            'permissions',
-            'breadcrumb',
-            'preview',
-            'favorites',
-            'subscribedNotifications',
-            'thumbnail',
-            'renditions',
-            'pendingTasks',
-            'runnableWorkflows',
-            'runningWorkflows',
-            'collections',
-            'audit',
-            'subtypes',
-            'tags',
-            'publications'
-          ].join(',');
+        return (Nuxeo.UI && Nuxeo.UI.config && Nuxeo.UI.config.enrichers) ||
+          {
+            document: [
+              'hasContent',
+              'firstAccessibleAncestor',
+              'permissions',
+              'breadcrumb',
+              'preview',
+              'favorites',
+              'subscribedNotifications',
+              'thumbnail',
+              'renditions',
+              'pendingTasks',
+              'runnableWorkflows',
+              'runningWorkflows',
+              'collections',
+              'audit',
+              'subtypes',
+              'tags',
+              'publications'
+            ],
+            blob: 'appLinks'
+          };
       },
 
       _computeHeaders: function() {
-        var fProps = (Nuxeo.UI && Nuxeo.UI.config && Nuxeo.UI.config.fetchedProperties) || 'properties';
-        return {
-          'X-NXfetch.directoryEntry': 'parent',
-          'X-NXfetch.document': 'lock,' + fProps,
-          'X-NXfetch.task': 'actors',
-          'X-NXtranslate.directoryEntry': 'label'
-        };
+
+        var headers = {
+          'translate-directoryEntry': 'label',
+        }
+
+        var fetch = Nuxeo.UI && Nuxeo.UI.config && Nuxeo.UI.config.fetch;
+        fetch = fetch|| {document: ['properties']};
+        // ensure we fetch lock details
+        fetch.document = fetch.document || [];
+        fetch.document.push('lock');
```

Comment:
We should double check `lock` isn't already there:
```suggestion
        if (!fetch.document.includes('lock')) {
          fetch.document.push('lock');
        }
```